index.html

```diff
@@ -1238,6 +1238,192 @@
+
+   <li><p>Add an entry to <var>result</var> called "<code>items</code>" whose
+   value is the array <var>items</var>.</p></li>
+
+   <li><p>Add an entry to <var>result</var> called "<code>@context</code>" whose value is the following object:</p>
+     <pre class="html">{ "@vocab" : "" }</pre></li>
+
+   <li><p>Return the result of serializing <var>result</var> to JSON. [[JSON]]</p></li>
+
+  </ol>
+
+  <p class="note">This algorithm returns an object with a single property that is an array, instead
+  of just returning an array, so that it is possible to extend the algorithm in the future if
+  necessary.</p>
+
+  <p>When the user agent is to <dfn>get the object</dfn> for an item <var>item</var>,
+  potentially together with a list of elements <var>memory</var>, it must run the following
+  substeps:</p>
+
+  <ol>
+
+   <li><p>Let <var>result</var> be an empty object.</p></li>
+
+   <li><p>If no <var>memory</var> was passed to the algorithm, let <var>memory</var> be an empty list.</p></li>
+
+   <li><p>Add <var>item</var> to <var>memory</var>.</p></li>
+
+   <li><p>If the <var>item</var> has any <a>item types</a>, add an entry to <var>result</var>
+   called "<code>@type</code>" whose value is an array listing the
+   <a>item types</a> of <var>item</var>, in the order they were specified on the
+   <code><a>itemtype</a></code> attribute.</p>
+
+   </li><li><p>If the <var>item</var> has a <a>global identifier</a>, add an entry to
+   <var>result</var> called "<code>@id</code>" whose value is the <a>global
+   identifier</a> of <var>item</var>.</p></li>
+
+   <li><p>Let <var>properties</var> be an empty object.</p></li>
+
+   <li>
+
+    <p>For each element <var>element</var> that has one or more <a>property names</a>
+    and is one of <a data-lt="item properties">the properties of the item</a>
+    <var>item</var>, in the order those elements are given by the algorithm that returns
+    <a data-lt="item properties">the properties of an item</a>, run the following substeps:</p>
+
+    <ol>
+
+     <li><p>Let <var>value</var> be the
+     <a>property value</a> of <var>element</var>.</p></li>
+
+     <li><p>If <var>value</var> is an <a data-lt="concept item">item</a>, then:
+     If <var>value</var> is in <var>memory</var>, then let <var>value</var> be
+     the string "<code>ERROR</code>". Otherwise, <a>get the object</a> for
+     <var>value</var>, passing a copy of <var>memory</var>, and then replace <var>value</var>
+     with the object returned from those steps.</p></li>
+
+     <li>
+
+      <p>For each name <var>name</var> in <var>element</var>'s
+      <a>property names</a>, run the following substeps:</p>
+
+      <ol>
+
+       <li><p>If there is no entry named <var>name</var> in <var>result</var>,
+       then add an entry named <var>name</var> to <var>result</var> whose
+       value is an empty array.</p></li>
+
```

Comment:
This looses the fact that the value may be a URL relationship, which should be encoded as `{"@id": "..."}`; alternatively, to simplify the syntax, and entry in the item-specific `@context` could be created for `name`, with `{"@type": "@id"}`.